cmake/CPM.cmake

```diff
@@ -323,6 +310,18 @@ function(CPMAddPackage)
       list(APPEND CPM_ARGS_UNPARSED_ARGUMENTS DOWNLOAD_COMMAND "${CMAKE_COMMAND}")
       set(PACKAGE_INFO "${download_directory}")
     else()
+      # If GIT_SHALLOW is explicitly specified, hornor the value.
+      # Otherwise, enable shallow clone only when GIT_TAG is not a commit hash.
+      # Our guess may not be accurate, but it should guarantee no commit hash get mis-detected.
+      if (DEFINED CPM_ARGS_GIT_SHALLOW)
+        list(APPEND CPM_ARGS_UNPARSED_ARGUMENTS GIT_SHALLOW ${CPM_ARGS_GIT_SHALLOW})
+      else()
```

Comment:
This part should still be considered in the default case (e.g. in line 233), as we still want to allow users to set `GIT_SHALLOW` explicitly in any case.